app/controllers/ReservationController.java

```diff
@@ -2,84 +2,63 @@
+import services.TableService;
 
+import javax.inject.Inject;
+import java.sql.Date;
 import java.sql.Time;
+import java.sql.Timestamp;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
-import java.util.Date;
 import java.util.List;
 
 /**
  * Created by lapa on 5/7/17.
  */
 public class ReservationController extends Controller {
 
+    private ReservationService reservationService;
+
+    @Inject
+    public void setReservationService(ReservationService reservationService) {
+        this.reservationService = reservationService;
+    }
+
+
     @Transactional(readOnly = true)
     public Result checkForTable() {
-//        Form<ReservationResponse> boundForm = ReservationResponse.reservationResponseForm.bindFromRequest();
-//
-//    int restaurantId = boundForm.get().getRestaurantId();
-//
-//    Restaurant restaurant = Restaurant.getRestaurantById(restaurantId);
-//
-//    String restaurantName = restaurant.getName();
-//
-//    String restaurantImage = restaurant.getImage();
-//
-//    Date date = boundForm.get().getDate();
-//
-//    String time = boundForm.get().getTime();
-//
-//    int people = boundForm.get().getPeople();
-//
-//    int numOfFreeTables = RestaurantTable.getTables(restaurantId, people).size();
-//
-//
-//        if(restaurantName == null){
-//                Logger.info("987987987987987987987");
-//            int restaurantId1 = 5;
-//            String restaurantName1 = "L A P A";
-//            String restaurantImage1 = "http://ceka.ba/images/ceka1-b.jpg";
-//            Date date1 = new Date();
-//            String time1 = "02:14 PM";
-//            int people1 = 7;
-//            int numOfFreeTables1 = 22;
-//            List<String> bestTimes1 = new ArrayList<>();
-//            bestTimes1.add(time.toString());
-//            bestTimes1.add("10:30 AM");
-//            bestTimes1.add("11:00 AM");
-//            bestTimes1.add("11:30 AM");
-//            ReservationResponse reservationResponse1 = new ReservationResponse(numOfFreeTables1,bestTimes1,restaurantId1,restaurantName1, restaurantImage1,date1,people1,time1);
-//
-//            return ok(Json.toJson(reservationResponse1));
-//                }
-//
-//
-//
-//
-//        List<String> bestTimes = new ArrayList<>();
-//        bestTimes.add(time.toString());
-//        bestTimes.add("10:30 AM");
-//        bestTimes.add("11:00 AM");
-//        bestTimes.add("11:30 AM");
-//
-//        SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy ");
-//        String d = sdf.format(date);
-//
-//        ReservationResponse reservationResponse = new ReservationResponse(numOfFreeTables,bestTimes,restaurantId,restaurantName, restaurantImage,date,people,time);
-//
-//
-//        return ok(Json.toJson(reservationResponse));
-//
-        return ok();
+
+        Form<Reservation> boundForm = Reservation.reservationForm.bindFromRequest();
+
+    Long restaurantId = Long.valueOf(boundForm.bindFromRequest().field("restaurantId").value());
```

Comment:
We should probably change this API, so id of restaurant is passed as path variable, so route should be `/api/v1/restaurants/:id/checkReservation` (where id is restaurant id).
Then this API method would have restaurantId as argument, so you don't have to bind it from request body.